### ISSUE TITLE
Define new textures based on vanilla ones

### ISSUE THREAD

**Issue by jaythebusinessgoose** [spelunky-fyi/overlunky#186]
Currently, I can't seem to get this to work:

```
local original_texture_definition = get_texture_definition(TEXTURE.DATA_TEXTURES_HUD_TEXT_0)
local texture_definition = TextureDefinition.new()
texture_definition.texture_path = original_texture_definition.texture_path
texture_definition.width = original_texture_definition.width
texture_definition.height = original_texture_definition.height
texture_definition.tile_width = 128
texture_definition.tile_height = 64
local texture = define_texture(texture_definition)
```

The only way this works is if I create a copy of the hud_text image and put it in my Data/Textures folder; it's not reading it from the vanilla game textures.

**Comment by Mr-Auto** [comment]
should be 
```lua
local texture_definition = get_texture_definition(TEXTURE.DATA_TEXTURES_HUD_TEXT_0)
texture_definition.tile_width = 128
texture_definition.tile_height = 64
local texture = define_texture(texture_definition)
```

`get_texture_definition` returns returns copy of the data, not pointer to game data, and for texture definition to work you need all the variables https://github.com/spelunky-fyi/overlunky/blob/main/docs/script-api.md#TextureDefinition

**Comment by jaythebusinessgoose** [comment]
Doing that immediately crashes when trying to define_texture.

**Comment by Mr-Auto** [comment]
that's question to @Malacath-92 then, why does it crash when defining new variant of a texture that exists in the game

**Comment by Malacath-92** [comment]
I'm quite certain it does not crash the game as I am doing that in one of my unreleased mods. I will be able to verify that once the tools are up and running again.

**Comment by jaythebusinessgoose** [comment]
Would you mind sharing a snippet of what you do so I can try it in my env?

Even something as simple as this is crashing for me:
```
local ltd = get_texture_definition(TEXTURE.DATA_TEXTURES_MENU_HEADER_0)
local lt = define_texture(ltd)
```

**Comment by Malacath-92** [comment]
I think the issue is that you have to prepend `Data/Textures/` to the path, since the vanilla paths don't usually include that but the API expects you to include it.

**Comment by Malacath-92** [comment]
Should be fixed by bc114c7

**Comment by Malacath-92** [comment]
And then cb02626 should fix the bug with redefining the same vanilla texture again that I made in the previous commit
